### PR TITLE
Add java dataset implemenation

### PR DESCRIPTION
**Pull request by mahmoudhanafy**

### CODE REVIEW

**Comment by mahmoudhanafy** [comment]
I'm wondering how it passes the tests here and fail at my machine.

File: src/test/1.6/java/com/holdenkarau/spark/testing/SampleJavaDatasetTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            @Override
+            public Human call(Human v) throws Exception {
+                v.setName("hello");
+                return v;
+            }
+        }, Encoders.bean(Human.class));
+
+        mapped.show(); // this gives exception
+
+        throw new RuntimeException("failed");
```
Comment (mahmoudhanafy) [inline src/test/1.6/java/com/holdenkarau/spark/testing/SampleJavaDatasetTest.java]: @holdenk I throw RuntimeException and it passed !

Comment (holdenk) [inline src/test/1.6/java/com/holdenkarau/spark/testing/SampleJavaDatasetTest.java]: thats weird...

File: src/test/1.6/scala/com/holdenkarau/spark/testing/SampleDatasetTest.scala [hunk 2]
```diff
@@ -21,21 +21,6 @@ class SampleDatasetTest extends DatasetSuiteBase {
     equalDatasets(persons, persons)
   }
   
-  test("unequal same schema, different classes") {
```
Comment (holdenk) [inline src/test/1.6/scala/com/holdenkarau/spark/testing/SampleDatasetTest.scala]: Why did we get rid of this?

Comment (mahmoudhanafy) [inline src/test/1.6/scala/com/holdenkarau/spark/testing/SampleDatasetTest.scala]: I can't do that in java. I can't get the class tag in java. 
I may add this logic only to Scala and prevent it in java.

Comment (holdenk) [inline src/test/1.6/scala/com/holdenkarau/spark/testing/SampleDatasetTest.scala]: ah hmmm... ok

**Comment by holdenk** [comment]
This looks good, and fixes a rather important bug. I'll merge and publish 0.3.1 soon unless we find any other bugs we should fix first.

**Pull request merged**